modules/build/src/test/scala/scala/build/tests/BuildTests.scala

```diff
@@ -917,11 +918,32 @@ abstract class BuildTests(server: Boolean) extends munit.FunSuite {
           |}
           |""".stripMargin
     )
-    inputs.withBuild(baseOptions, buildThreads, bloopConfigOpt) { (_, _, maybeBuild) =>
-      expect(maybeBuild.exists(_.success))
-      val build = maybeBuild.toOption.flatMap(_.successfulOpt).getOrElse(sys.error("cannot happen"))
-      val cp    = build.fullClassPath
-      expect(cp.length == 1) // no scala-library, only the class directory
+    val options = baseOptions.copy(
+      scalaOptions = baseOptions.scalaOptions.copy(
+        supportedScalaVersionsUrl =
+          Some("abcd://nope"), // should not be checked, as we're a pure Java project
+        ignoreSupportedScalaVersionsErrors = Some(false)
+      )
+    )
+    val gotExpectedError =
+      try {
+        options.scalaParams
+        false
+      }
+      catch {
+        case e if e.getCause.isInstanceOf[ArtifactError.DownloadError] =>
```

Comment:
If we want to avoid downloading supported Scala Version, why not change type `ScalaOptions.scalaVersionsUrl` to `Option[String]`. And download supportedScalaVersion only when variable `supportedScalaVersionsUrl ` have some value.

Instead of catching a download error, we just might not download the supported Scala Versions.